plugins/tickets.py

```diff
@@ -0,0 +1,2118 @@
+def resolve_ticket(msg, args, rewind=False) -> Ticket:
+    """
+    Resolves a ticket from the given message and command args, if possible.
+    Ticket is extracted from either the referenced message or the first arg.
+    """
+    ticket = None
+    if ref := msg.reference:
+        if (ref_msg := ref.resolved) and isinstance(ref_msg, discord.Message):
+            if ref_msg.author == client.user and ref_msg.embeds:
+                embed = ref_msg.embeds[0]
+                if (name := embed.author.name) and name.startswith("Ticket #"):
+                    ticket_id = int(name[8:].split(' ', maxsplit=1)[0])
+                    ticket = get_ticket(ticket_id)
+    if ticket is None:
+        rewind_pos = args.pos
+        ticketarg = args.next_arg()
+        if ticketarg is not None and isinstance(ticketarg, commands.StringArg):
+            maybe_id = int(ticketarg.text)
+            if maybe_id < 2147483647:  # Hack to differentiate msgid from tid
+                tickets = fetch_tickets_where(id=maybe_id)
+            else:
+                tickets = fetch_tickets_where(list_msgid=maybe_id)
+            ticket = next(tickets, None)
+        if rewind and ticket is None:
+            args.pos = rewind_pos
+    return ticket
+
+
+def summarise_tickets(*tickets, title="Tickets", fmt=None):
+    """
+    Create paged embeds of ticket summaries from the provided list of tickets.
+    """
+    if not tickets:
+        return None
+
+    lines = [ticket.summary(fmt=fmt) for ticket in tickets]
+    blocks = ['\n'.join(lines[i:i+10]) for i in range(0, len(lines), 10)]
+    page_count = len(blocks)
+
+    embeds = (
+        discord.Embed(description=blocks[i], title=title)
+        for i in range(page_count)
+    )
+
+    if page_count > 1:
+        embeds = (
+            embed.set_footer(text="Page {}/{}".format(i+1, page_count))
+            for i, embed in enumerate(embeds)
+        )
+
+    return embeds
+
+
+Page = namedtuple('Page', ('content', 'embed'), defaults=(None, None))
+
+
+async def pager(dest: discord.abc.Messageable, pages):
+    """
+    Page a sequence of pages.
+    """
+    _next_reaction = '⏭️'
+    _prev_reaction = '⏮️'
+    _all_reaction = '📜'
+    reactions = (_prev_reaction, _all_reaction, _next_reaction)
+
+    pages = list(pages)
+
+    # Sanity check
+    if not pages:
+        raise ValueError("Cannot page with no pages!")
+
+    # Send first page
+    msg = await dest.send(**pages[0]._asdict())
+
+    if len(pages) == 1:
+        return
+
+    # Add reactions
+    for r in reactions:
+        await msg.add_reaction(r)
+
+    index = 0
+    while True:
+        try:
+            with ReactionMonitor(
+                message_id=msg.id,
+                event='add',
+                filter=lambda _, p: (
+                    str(p.emoji) in reactions
+                    and p.user_id != msg.guild.me.id
+                ),
+                timeout_each=120
+            ) as mon:
+                _, payload = await mon
```

Comment:
Instead of creating a ReactionMonitor every time, you can iterate over `mon` to get multiple events

P.S. currently you will miss reactions that arrive during any of the awaits you have in this `with` block.